hlapipeline/alignimages.py

```diff
@@ -3,65 +3,173 @@
+MIN_CROSS_MATCHES = 3
+MIN_FIT_MATCHES = 6
 
 # Module-level dictionary contains instrument/detector-specific parameters used later on in the script.
 detector_specific_params = {"acs":
                                 {"hrc":
-                                     {"fwhmpsf": 0.073},  # TODO: Verify value
+                                     {"fwhmpsf": 0.073,
+                                      "classify": True,
+                                      "threshold": None},
                                  "sbc":
                                      {"fwhmpsf": 0.065,
-                                     "threshold":10,
-                                     "classify":False
-                                      },  # TODO: Verify value
+                                      "classify": False,
+                                      "threshold": 10},
                                  "wfc":
-                                     {"fwhmpsf": 0.076}},  # TODO: Verify value
+                                     {"fwhmpsf": 0.076,
+                                      "classify": True,
+                                      "threshold": None}},  # TODO: Verify ACS fwhmpsf values
                             "wfc3":
                                 {"ir":
-                                     {"fwhmpsf": 0.14},
+                                     {"fwhmpsf": 0.14,
+                                      "classify": False,
+                                      "threshold": None},
                                  "uvis":
-                                     {"fwhmpsf": 0.076}}}
+                                     {"fwhmpsf": 0.076,
+                                      "classify": True,
+                                      "threshold": None}}}
 # ----------------------------------------------------------------------------------------------------------------------
 
 
-def main(input_list):
+def check_and_get_data(input_list):
+    """Verify that all specified files are present. If not, retrieve them from MAST.
+
+    Parameters
+    ----------
+    imglist : list
+        List of one or more calibrated fits images that will be used for catalog generation.
+
+    Returns
+    =======
+    input_file_list : list
+        list of full filenames
+
+    """
+    totalInputList=[]
+    for input_item in input_list:
+        if input_item.endswith("0"): #asn table
+            totalInputList += aqutils.retrieve_observation(input_item)
+
+        else: #single file rootname.
+            fitsfilename = glob.glob("{}_flc.fits".format(input_item))
+            if not fitsfilename:
+                fitsfilename = glob.glob("{}_flt.fits".format(input_item))
+            fitsfilename = fitsfilename[0]
+
+            if not os.path.exists(fitsfilename):
+                imghdu = fits.open(fitsfilename)
+                imgprimaryheader = imghdu[0].header
+                totalInputList += aqutils.retrieve_observation(imgprimaryheader['ASN_ID'].strip())
```

Comment:
For true singletons, ``ASN_ID`` will be blank (if present, at all).  It is not clear whether that possibility has been taken into account here so that an Exception is not raised.

Reply:
Issue resolved in revision [
9882e5d](https://github.com/spacetelescope/hlapipeline/pull/36/commits/9882e5dc1b8c0bda62387a44f5b891a154e52119)